src/Generator.cpp

```diff
@@ -97,29 +99,520 @@ Outputs compute_outputs(const Target &target,
     return output_files;
 }
 
+Argument to_argument(const Internal::Parameter &param) {
+    Expr def, min, max;
+    if (!param.is_buffer()) {
+        def = param.get_scalar_expr();
+        min = param.get_min_value();
+        max = param.get_max_value();
+    }
+    return Argument(param.name(),
+        param.is_buffer() ? Argument::InputBuffer : Argument::InputScalar,
+        param.type(), param.dimensions(), def, min, max);
+}
+
+std::pair<int64_t, int64_t> rational_approximation(double d) {
+    if (std::isnan(d)) return {0, 0};
+    if (!std::isfinite(d)) return {(d < 0) ? -1 : 1, 0};
+    // TODO: fix this abomination to something more intelligent
```

Comment:
This is a fun topic:
https://www.youtube.com/watch?v=CaasbfdJdJg

I'll open a small PR with my solution. It's overkill.